zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -627,7 +627,9 @@ public void putNoteInterpreterSettingBinding(String noteId,
 
   public void removeNoteInterpreterSettingBinding(String noteId) {
     synchronized (interpreterSettings) {
-      List<String> settingIds = interpreterBindings.remove(noteId);
+      @SuppressWarnings("unchecked")
+      List<String> settingIds = interpreterBindings.containsKey(noteId) ?
+          interpreterBindings.remove(noteId) : Collections.EMPTY_LIST;
```

Comment:
I think `Collections,emptyList();` is better here (will remove the `@SuppressWarnings("unchecked")`)